ci/base-tests.sh

```diff
@@ -24,21 +24,21 @@ cd clippy_lints && cargo test && cd ..
 cd rustc_tools_util && cargo test && cd ..
 # check that the lint lists are up-to-date
 ./util/update_lints.py -c
-mkdir -p ~/rust/cargo/bin
-cp target/debug/cargo-clippy ~/rust/cargo/bin/cargo-clippy
-cp target/debug/clippy-driver ~/rust/cargo/bin/clippy-driver
-rm ~/.cargo/bin/cargo-clippy
+
+CLIPPY="`pwd`/target/debug/cargo-clippy clippy"
 # run clippy on its own codebase...
-PATH=$PATH:~/rust/cargo/bin cargo clippy --all-targets --all-features -- -D clippy::all -D clippy::internal
+${CLIPPY} --all-targets --all-features -- -D clippy::all -D clippy::internal
 # ... and some test directories
-cd clippy_workspace_tests && PATH=$PATH:~/rust/cargo/bin cargo clippy -- -D clippy::all && cd ..
-cd clippy_workspace_tests/src && PATH=$PATH:~/rust/cargo/bin cargo clippy -- -D clippy::all && cd ../..
-cd clippy_workspace_tests/subcrate && PATH=$PATH:~/rust/cargo/bin cargo clippy -- -D clippy::all && cd ../..
-cd clippy_workspace_tests/subcrate/src && PATH=$PATH:~/rust/cargo/bin cargo clippy -- -D clippy::all && cd ../../..
-cd clippy_dev && PATH=$PATH:~/rust/cargo/bin cargo clippy -- -D clippy::all && cd ..
-cd rustc_tools_util/ && PATH=$PATH:~/rust/cargo/bin cargo clippy -- -D clippy::all && cd ..
+CWD_OLD=`pwd`
+for dir in clippy_workspace_tests clippy_workspace_tests/src clippy_workspace_tests/subcrate clippy_workspace_tests/subcrate/src clippy_dev rustc_tools_util
+do
+    cd ${dir}
+    ${CLIPPY} -- -D clippy::all
+    cd ${CWD_OLD}
```

Comment:
Maybe `pushd` and `popd` since Travis uses bash?

Reply:
I don't understand.
The bash script and cd'ing inside it should work just fine on travis?

Reply:
Ah no, it will definitely work. `pushd` & `popd` was just a suggestion for readability and to me it looked like an easy "-1 LoC optimization" :D.

Reply:
Mh, I am not familiar with `pushd` and `popd` and have never used them so I would rather keep it the current way, but thanks for the suggestion  :)

Reply:
Since this is just going back and forth between two directories, I think `pushd` and `popd` are not really needed here. But I the `cd ${CWD_OLD}` can be replaced with `cd -`, so we don't need to store `CWD_OLD`.

Reply:
:+1:  updated